content/v2.0/application/config.md

```diff
@@ -0,0 +1,102 @@
+---
+title: Config
+order: 50
+---
+
+You can configure various aspects of your Hanami app using `config` in your app class.
+
+```ruby
+# config/app.rb
+
+module Bookshelf
+  class App < Hanami::App
+    config.inflections do |inflections|
+      inflections.acronym "WNBA"
+    end
+  end
+end
+```
+
+You can also configure your app differently in different environments.
+
+```ruby
+# config/app.rb
+
+module Bookshelf
+  class App < Hanami::App
+    environment(:production) do
+      require "my_custom_middleware"
+      config.middleware.use MyCustomMiddleware
+    end
+  end
+end
+```
+
+See the [environments guide](/v2.0/application/environments) for more detail.
+
+Individual slices may also be configured. See the [slices guide](/v2.0/architecture/slices) for more detail.
+
+See below for a description of all available `config` methods.
+
+## General
+
+### `inflections`
+
+Along with `inflector`, customizes your your app's string inflection rules. See the [inflector guide](/v2.0/application/inflector) for more detail.
+
+### `slices`
+
+Specifies the slices to load. See the [slices guide](/v2.0/architecture/slices) for more detail.
+
+## Code loading
+
+### `root`
+
+Sets the root for the app or slice. For the app, defaults to `Dir.pwd`. For slices, defaults to the slice's name under the `/slices/` directory.
+
+The root is used for locating the code to be loaded by the app. See the [autoloading guide](/v2.0/application/autoloading) and the [containers and components guide](/v2.0/architecture/containers) for more detail.
+
+### `shared_app_component_keys`
+
+Sets the keys for the app components to be automatically imported into each slice. See the [slices guide](/v2.0/architecture/slices) for more detail.
+
+### `no_auto_register_paths`
+
+Sets an array of paths (relative to the root of the app or any slice) to be excluded from component auto-registration. Defaults to `["entities"]`. See the [containers and components guide](/v2.0/architecture/containers) for more detail.
+
+## Router
+
+### `base_url`
+
+Sets the base URL for the app's web server. This is used when building URLs for named routes. Defaults to `"http://0.0.0.0:2300"`. See the [routes guide](/v2.0/routes) for more detail.
+
+### `middleware`
+
+Configures the Rack middleware stack to be used by the app's router.
+
+Use `use` to append a middlware.
+
+```ruby
+# config/app.rb
+config.middlware.use MyMiddleware, "middlware", "args", "here"
+```
+
+Use the `before:` or `after:` options to insert a middleware at a particular point in the stack:
+
+```ruby
+# config/app.rb
+config.middleware.use MyMiddleware, before: AlreadyAddedMiddleware
+```
+
+Defaults to an empty middleware stack. If you configure actions to use the `:json` format, then `Hanami::Middleware::BodyParser` will be added automatically and configured to parse JSON request bodies.
```

Comment:
"Defaults to an empty middleware stack." <- was something missing here?

Reply:
@andrewcroome That sentence, if you want to expand it, would read "This setting defaults to an empty middleware stack".

I was just using a condensed sentence structure to avoid repeating "This setting" in front of all such sentences, e.g. like I do here:

> Sets the base URL for the app's web server. This is used when building URLs for named routes. Defaults to `"http://0.0.0.0:2300"`.

Happy to take your advice on how to make sure this is clear at all times.

Reply:
Pushed a commit to the end here to hopefully make things clearer. I'll merge this PR now but I'd be happy if you wanted to go in later and adjust things!